pkg/daemon/daemon.go

```diff
@@ -37,13 +37,14 @@ const (
 type NewDaemonOpt func(d *Daemon) error
 
 type States struct {
-	// Generated by daemons manager as a unique to identify a nydusd
+	// A unique ID generated by daemon manager to identify the nydusd instance.
 	ID          string
 	ProcessID   int
 	APISocket   string
 	LogDir      string
 	LogLevel    string
 	LogToStdout bool
+	DaemonMode  config.DaemonMode
```

Comment:
Do we need to handle the compatibility if the newer nydus-snapshotter is loading old records in DB without `DamonMode`  ?

Reply:
Yes, we need to migrate the database, that's on the todo list.

Reply:
Currently it works, but may get broken in future easily. So the safe way is to convert records in databases. `Rafs` will face the same situation in coming PRs, so let's delay support of data migration.

Reply:
Um. I tend to tolerate the DB compatibility within this PR rather than make it a future work.
For the compatibility, I suppose it wouldn't be very complicated. Nydus-sanpshotter already has an experience on how to migrate different record formats in function `tryTranslateRecords()`. We can simply assign a DaemonMode by comparing daemon `Mountpoint` and `HostRootMount` and rewrite the daemon's record of state

Reply:
Added a patch to migrate the database.